### PR TITLE
switched to yarn

### PR DESCRIPTION
**Pull request by artlowel**
This PR connects to #65 

It replaces npm with yarn to manage the dependencies of the project

We shouldn't merge this until at least one person using linux and one person using windows has verified that it works.

## Note: install using npm
I opted to add yarn to the global install script and install it using npm, that means people don't need to install yarn manually in advance, and our quick start guide can remain short, but [it isn't recommended by the yarn devs](https://yarnpkg.com/en/docs/install#install-via-npm) - they advise you to use the installer. On my system I installed both, and the installer version gets priority over the node version, but that's likely up to the order they're added to `$PATH` and is by no means a guarantee.

What do you think? Should we install it automatically, or have people install it manually?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,14 +53,14 @@ Not sure where to start? watch the training videos linked in the [Introduction t
 You can find more information on the technologies used in this project (Angular 2, Typescript, Angular Universal, RxJS, etc) on the [DuraSpace wiki](https://wiki.duraspace.org/display/DSPACE/DSpace+7+UI+Technology+Stack)
 
 ## Requirements
-* [Node.js](https://nodejs.org) and [npm](https://www.npmjs.com/)
-* Ensure you're running node >= `v5.x` and npm >= `v3.x`
+* [Node.js](https://nodejs.org) and [yarn](https://yarnpkg.com)
```
Comment (tdonohue) [inline README.md]: NPM is likely still required, correct?

Comment (artlowel) [inline README.md]: Yes that shouldn't have been removed. I'll fix it

**Comment by hardyoyo** [comment]
I can verify this works as expected on Linux, however, one warning seemed a bit concerning:

```
> spawn-sync@1.0.15 postinstall /usr/local/lib/node_modules/yarn/node_modules/spawn-sync
> node postinstall

- mimic-fn@1.1.0 node_modules/yarn/node_modules/mimic-fn
/usr/local/lib
├─┬ angular-cli@1.0.0-beta.28.3 
│ ├─┬ cssnano@3.10.0
│ │ └─┬ postcss-merge-rules@2.1.2
│ │   └── postcss-selector-parser@2.2.3 
│ ├── rimraf@2.6.1 
│ └── UNMET PEER DEPENDENCY rxjs@^5.0.1
... /* time passes */ ...
npm WARN @angular/core@2.4.8 requires a peer of rxjs@^5.0.1 but none was installed.

```

observed while running `yarn run global`

**Comment by artlowel** [comment]
@hardyoyo That issue isn't specific to yarn, it's that the `angular-cli` package we had in the `global` has since been renamed to `@angular/cli` and of course the old `angular-cli` package is no longer updated, so is no longer compatible with the dependencies in our project which _have_ been updated since
I changed it to the new name that fixes the problem for me.

**Comment by atarix83** [comment]
Hi,
I finally tried yarn on windows and it all works properly.
I agree with global installation of yarn, it's the quickest solution.

**Comment by artlowel** [comment]
>I agree with global installation of yarn, it's the quickest solution.
I've changed my mind on this. Yes it is slightly quicker to do the initial install, but in the full install docs we _do_ recommend to install yarn properly. And if you do both, you can end up with an uncertainty about which version is being used. 
I have therefore added yarn as a prerequisite to the quick install as well, and removed it from the global script.

**Pull request merged**